fossunited/fossunited/doctype/grants_funding_directory/test_grants_funding_directory.py

```
@@ -1,8 +1,203 @@
+        url = self.upload_temp(bad)
+
+        doc = frappe.get_doc(
+            {
+                "doctype": "Grants Funding Directory",
+                "funding_json": url,
+                "name": "Real Invalid FundingTest",
+            }
+        )
+
+        # The real API should reject this
+        with self.assertRaises(Exception):
```

Ruff (0.14.5)

Do not assert blind exception: `Exception`

(B017)